etna/transforms/math/lags.py

```diff
@@ -226,11 +240,11 @@ def _estimate_shift(self, df: pd.DataFrame, feature_name: str) -> int:
         if not self._auto:
             return self.lag  # type: ignore
 
-        if self._exog_last_date is None or self._freq is None:
+        if self._exog_last_timestamp is None or self._freq is _DEFAULT_FREQ:
```

Comment:
We can remove `self._freq` from this check and test if transform is fitted based on `self._exog_last_timestamp` only

Reply:
I think both these checks added only to explain to type checked that both this values exist.

Reply:
Let's remove `self._freq` from check